docs/workflows/accessing-payload-data.md

```diff
@@ -0,0 +1,330 @@
+{{data.aBooleanProp}} => true
+{{data.anUndefinedProp}} => (prints nothing)
+{{data.aMissingProp}} => (prints nothing)
+{{data.anArray.[1]}} => Goodbye world
+{{data}} => [object Object]
+
+{{data.anArray}} => 44,Goodbye world,false,[object Object]
+(all items in the array concatenated by commas)
+
+{{data.anArray.[1].[0]}} => G
+(the character at index 0 of the item at index 1 of the array)
+
+{{data.anArray.0}} => (prints nothing)
+(0 is not a property of the array; should be [0] to access by index)
+
+{{data.anArray.[16]}} => (prints nothing)
+(index does not exist on the array.)
+
+{{#if data.aNumberProp}}The number is {{data.aNumberProp}}{{/if}}
+=> The number is 42
+
+{{#if data.aMissingProp}}The number is {{data.aMissingProp}}{{/if}}
+=> (prints nothing. The condition fails.)
+
+{{#if data.aFalsyProp}}Truth!{{else}}Fiction!{{/if}}
+=> Fiction! (0 is falsy so the condition falls to the else statement.)
+
+{{#each data.anArray}}{{@index}}: {{this}}; {{/each}}
+=> 0: 44; 1: Goodbye world; 2: false; 3: [object Object];
+
+{{#each data.anObject}}{{@key}}: {{this}}; {{/each}}
+=> aNestedArray: World Star!,20,[object Object]; aNestedNumber: 56; aNestedString: Wally World;
+
+{{#each data.aMissingObject}}{{@key}}: {{this}} -- {{else}}No object!{{/each}}
+=> No object!
+
+{{#gt data.aNumberProp 40}}First number is greater{{/gt}}
+=> First number is greater
+
+{{#lte data.aNumberProp 10}}Second number is greater{{/gt}}
+=> (prints nothing)
+
+{{#eq data.aStringProp 'hello world'}}Equal!{{else}}Not equal{{/eq}}
+=> Not equal (cases don't match)
+
+{{#eq (lower data.aStringProp) 'hello world'}}Equal!{{else}}Not equal{{/eq}}
+=> Equal! (cases match after running through the "lower" subexpression)
+
+{{format data}} => (the stringified object)
+{{format data.aDecimal ',.5'}} => 1,234.6
+{{encodeURIComponent data.aStringProp}} => Hello%20world
+
+{{#eq (format data.aJavascriptDateObject 'MMMM') 'September'}}Ba de ya!{{else}}Play something else{{/if}}
+=> Ba de ya!
+```
+
+Remember that, when checking strings for equality in Indicator block expressions and Conditional nodes, or when including a string template in a JSON object, you'll need to wrap the string template in quotes:
+
+## Expressions
+
+Sometimes it is necessary to evaluate our data and determine which of two (or many) paths to follow as a result of the evaluation, such as in the dashboard [Indicator block's expressions](/dashboards/indicator/) and the workflow [Conditional node](/workflows/logic/conditional/). In these cases we use **expressions**.
+
+An expression is a combination of string templates and [JavaScript operators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Expressions_and_Operators) that forms a valid JavaScript expression after parsing through the Handlebars engine. The expression returns a value, which is evaluated as [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) or [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy).
```

Comment:
this is not quite true - the math nodes do not force truth/false out of the result.  Also, it is more like "javascript-like" expressions, since there are tons of JS things you can do that would not be valid in expressions